ctapipe_io_lst/__init__.py

```diff
@@ -726,20 +740,17 @@ def __init__(self, file_list):
                 self._events_table[path] = self._file[path].Events
                 self._events[path] = next(self._file[path].Events)
 
-                # verify where the CameraConfig is present
-                if 'CameraConfig' in self._file[path].__dict__.keys():
+                if hasattr(self._file[path], 'CameraConfig'):
                     self._camera_config[path] = next(self._file[path].CameraConfig)
 
-                # for the moment it takes the first CameraConfig it finds (to be changed)
-                    if self.camera_config is None:
-                        self.camera_config = self._camera_config[path]
-
             except StopIteration:
                 pass
 
-        # verify that somewhere the CameraConfing is present
-        if self.camera_config is None:
+        # verify that we found a CameraConfig
+        if len(self._camera_config) == 0:
```

Comment:
it allows for a situation that some of the files are missing CameraConfig, or that the CameraConfig between different files are not matching, is it intended?
maybe check that the len of camera configs is the same as len of the file list

Reply:
The logic of the check has not changed. I just improved the way it was done. And yes, I think the intention was to allow missing camera configs in streams and only checking that at least one stream has a camera config.

Reply:
I'd say this is how it was intended and it's not really connected to this PR.